arraycontext/impl/pyopencl/fake_numpy.py

```diff
@@ -259,6 +259,24 @@ def equal(self, x, y):
     def not_equal(self, x, y):
         return rec_multimap_array_container(operator.ne, x, y)
 
+    def logical_or(self, x, y):
+        def _rec(ary1, ary2):
+            if ary1.dtype != np.bool_ or ary2.dtype != np.bool_:
```

Comment:
Doesn't pyopencl use `int8` for bools? Does this work as expected?

Reply:
Thanks, yep. It was wrong earlier. Just fixed pyopencl to generate kernels to support all types. See https://github.com/inducer/pyopencl/pull/679.